lib/experimental/fonts-api/class-wp-fonts-resolver.php

```diff
@@ -202,10 +202,20 @@ private static function get_settings() {
 				$settings            = static::set_tyopgraphy_settings_array_structure( $settings );
 			}
 
+			// Initialize the font families from settings if set and is an array, otherwise default to an empty array.
+			$settings_font_families = ( isset( $settings['typography']['fontFamilies']['theme'] ) && is_array( $settings['typography']['fontFamilies']['theme'] ) )
+				? $settings['typography']['fontFamilies']['theme']
+				: array();
```

Comment:
Thinking this can be outside (above) the loop as `$settings` is not changed in the loop, just a bit of optimization.

At the same time not sure what is the expected default value (and value type) of `$settings['typography']['fontFamilies']['theme']` when it is empty or not set. Same for `$variation['settings']['typography']['fontFamilies']['theme']`. Setting these to empty arrays when missing would return an empty array, is that expected?